shell/platform/tizen/tizen_window.h

```diff
@@ -6,71 +6,31 @@
 #ifndef EMBEDDER_TIZEN_WINDOW_H_
 #define EMBEDDER_TIZEN_WINDOW_H_
 
+#include "flutter/shell/platform/tizen/tizen_view_base.h"
+
 #include <cstdint>
 #include <string>
 #include <vector>
 
-#include "flutter/shell/platform/tizen/tizen_input_method_context.h"
-
 namespace flutter {
 
 class FlutterTizenView;
 
-class TizenWindow {
+class TizenWindow : public TizenViewBase {
  public:
-  struct Geometry {
-    int32_t left = 0, top = 0, width = 0, height = 0;
-  };
-
+  TizenWindow() = default;
   virtual ~TizenWindow() = default;
 
-  // Sets the delegate used to communicate state changes from window to view
-  // such as key presses, mouse position updates etc.
-  void SetView(FlutterTizenView* view) { view_ = view; }
-
-  // Returns the geometry of the current window.
-  virtual Geometry GetWindowGeometry() = 0;
-
-  // Set the geometry of the current window.
-  virtual void SetWindowGeometry(Geometry geometry) = 0;
-
-  // Returns the geometry of the display screen.
-  virtual Geometry GetScreenGeometry() = 0;
-
-  // Returns a valid pointer the platform object that rendering can be bound to
-  // by rendering backend.
-  virtual void* GetRenderTarget() = 0;
-
-  virtual void* GetRenderTargetDisplay() = 0;
+  virtual void* GetWindowHandle();
 
   virtual int32_t GetRotation() = 0;
 
-  // Returns the dpi of the screen.
-  virtual int32_t GetDpi() = 0;
-
-  virtual uintptr_t GetWindowId() = 0;
+  virtual void SetPreferredOrientations(const std::vector<int>& rotations);
 
-  virtual void* GetWindowHandle() = 0;
-
-  virtual void ResizeRenderTargetWithRotation(Geometry geometry,
-                                              int32_t degree) = 0;
-
-  virtual void SetPreferredOrientations(const std::vector<int>& rotations) = 0;
-
-  virtual void BindKeys(const std::vector<std::string>& keys) = 0;
-
-  virtual void Show() = 0;
-
-  // FIXME
-  // This is a temporary implementation that is only used by the window channel.
-  virtual void OnGeometryChanged(Geometry geometry) = 0;
-
-  TizenInputMethodContext* input_method_context() {
-    return input_method_context_.get();
-  }
+  std::string GetType() override { return "window"; };
```

Comment:
Can we use a dedicated enum instead of a string to represent the type?

Actually this kind of runtime type check is discouraged by [the style guide](https://google.github.io/styleguide/cppguide.html#Run-Time_Type_Information__RTTI_). We may need to either redesign our class hierarchy, or refactor the code so that the type check becomes unnecessary.

> Do not hand-implement an RTTI-like workaround. The arguments against RTTI apply just as much to workarounds like class hierarchies with type tags.

Reply:
I modified it to use enum.